packages/driver/src/cy/commands/xhr.coffee

```diff
@@ -403,6 +403,14 @@ module.exports = (Commands, Cypress, cy, state, config) ->
           ## aliases subject
           options.response = aliasObj.subject
 
+        url = getUrl(options)
+
+        urlString = url.toString()
+
+        ## https://github.com/cypress-io/cypress/issues/2372
+        if (decodedUrl = decodeURI(urlString)) and urlString != decodedUrl
```

Comment:
I think you need to handle malformed URIs here, at least I don't see this being handled before this code. `decodeURI` will throw on a malformed URI. 

Can you write a test when passing in `http://example.com/%E0%A4%A` to the url option? If not being handled here...it should be handled somewhere...

Reply:
Ooh, good point. It seems to throw if you percent-encode an unsupported codepoint: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors/Malformed_URI?utm_source=mozilla&utm_medium=firefox-console-errors&utm_campaign=default

I guess the good thing to do would be to just continue if decodeURI throws, assuming that because they have potentially "invalid codepoints" in the URI, they're not intending to use percent-encoding anyways and there's no need to warn.